src/controllers/soil_controller.py

```diff
@@ -0,0 +1,28 @@
+class SoilController:
+    def __init__(self):
+        self.soil_type = None
+        self.precipitation = None
+        self.temperature = None
+
+    def input_data(self, soil_type, precipitation, temperature):
+        self.soil_type = soil_type
+        self.precipitation = precipitation
+        self.temperature = temperature
+
+    def calculate_soil_conditions(self):
+
+        if self.soil_type == "sandy":
```

Comment:
Could we represent this as a dictionary instead i.e. 
soil_type = { clay: {'ph': 6.0, ''moisture_content_in_soil":0.05 * self.precipitation - 0.01 * self.temperature}
loamy: { } etc etc

This would make it easier to add more soil types in the future